compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -1769,9 +1801,12 @@ fn for_each_exported_symbols_include_dep<'tcx>(
     }
 }
 
-pub(crate) fn exported_symbols(tcx: TyCtxt<'_>, crate_type: CrateType) -> Vec<String> {
+pub(crate) fn exported_symbols(
+    tcx: TyCtxt<'_>,
+    crate_type: CrateType,
+) -> Vec<(String, SymbolExportKind)> {
     if let Some(ref exports) = tcx.sess.target.override_export_symbols {
-        return exports.iter().map(ToString::to_string).collect();
+        return exports.iter().map(|name| (name.to_string(), SymbolExportKind::Text)).collect();
```

Comment:
I removed the SymbolExportKind for exported_symbols change from my PR, but I believe your review comment https://github.com/rust-lang/rust/pull/142568#discussion_r2150403133 applies here too. This was the fixme I came up with: https://github.com/rust-lang/rust/commit/fc88eae9c346e99f311fd920760c7a6077853369

```rust
// FIXME use the correct export kind for this symbol. override_export_symbols
// can't directly specify the SymbolExportKind as it is defined in rustc_middle
// which rustc_target can't depend on.
```